tests/unittests/helpers.py

```diff
@@ -504,4 +506,18 @@ def __mock_assert_not_called(mmock):
             raise AssertionError(msg)
     mock.Mock.assert_not_called = __mock_assert_not_called
 
+
+def get_top_level_dir() -> Path:
+    """Return the path to the top cloudinit project directory
+
+    @return Path('<top-cloudinit-dir>')
+    """
+    return Path(cloudinit.__file__).parent.parent.resolve()
+
+
+def CloudinitDir(sub_path: str) -> str:
```

Comment:
Is there a reason we're going with this naming? Usually functions are lowercase and snake case. Additionally, I don't think the naming tells me enough to know if this refers to the top-level project directory or the `cloudinit` directory within it.

I also wonder how much we need this additional helper. To take a random example from this PR, is there really much difference between:
`CloudinitDir('tools/ds-identify')`
vs
`get_top_level_dir() / "tools/ds-identify"`

(with maybe a `str` cast when needed?)

Reply:
> Is there a reason we're going with this naming? Usually functions are lowercase and snake case. Additionally, I don't think the naming tells me enough to know if this refers to the top-level project directory or the cloudinit directory within it.

I was trying to think of something short that got the point across and that's what I landed on. Happy to change it to something better. I actually had it implemented as a class in a previous iteration, hence the style.

`get_cloud_init_project_dir()`?

`cloud_init_parent_dir()`?

I'm open to suggestions.

> I also wonder how much we need this additional helper. To take a random example from this PR, is there really much difference between:
CloudinitDir('tools/ds-identify')
> vs
> get_top_level_dir() / "tools/ds-identify"
>
> (with maybe a str cast when needed?)

Most of the locations that I've replaced so far actually expect to do `str` things (+ concatenation, encode('utf-8'), etc) to the returned object. If we did that we would have to cast many of the returned values, which is why I chose this approach.

[A previous iteration](https://github.com/holmanb/cloud-init/commit/b03b911e52deac1cf225f0a8b720a8afa81a130e#) actually subclassed `Path()` and implemented the magic methods required currently (`__add__`, `__addr__`, as well as `encode()`). But something in Python 3.5 was failing which drove me to just return a `str` from the function. (iirc it was join() in 3.5 standard library `os` functions requiring the objects to actually be `str` objects not just implement the magic methods, or something like that).

Reply:
Thanks, that explanation makes sense. 

As far as naming, I think "project" or "root" is clearer than "parent". `get_cloud_init_project_dir()` or `cloud_init_root_dir()`? Pick your poison.